File: SampleApps/FlutterHost/lib/duo_demo.dart
// Phase A5 — FlutterHost adapter demo (intentional R6 hits).
// Adapter smoke — calls DuoHarnessFlutter size/hinge APIs.
import '../../../Adapters/DuoHarnessFlutter/lib/duo_harness.dart';

Future<void> demoDuoHelpers() async {
  final insets = await DuoHarness.safeAreaInsets();
  final compact = await DuoHarness.isCompactWidth(400);
  final hinge = await DuoHarness.hingeFraction();
  final reserved = await DuoHarness.reservedInsets();
  DuoHarness.observeHinge().listen((_) {});
  assert(insets.top >= 0);
  assert(compact == true || compact == false);
  assert(hinge >= 0);
  assert(reserved.left >= 0);
}

File: SampleApps/RNHost/src/duoDemo.ts
// Phase A5 — RNHost adapter demo (intentional R6 hits).
// Adapter smoke — calls iPhoneDuoPrepRN size/hinge APIs.
import {
  hingeFraction,
  isCompactWidth,
  observeHinge,
  reservedInsets,
  safeAreaInsets,
} from "../../../Adapters/iPhoneDuoPrepRN/src";

export async function demoDuoHelpers(): Promise<void> {
  const insets = await safeAreaInsets();
  const compact = await isCompactWidth(400);
  const hinge = await hingeFraction();
  const reserved = await reservedInsets();
  observeHinge(() => {});
  if (insets.top < 0 || hinge < 0 || reserved.left < 0) {
    throw new Error("unexpected negative inset");
  }
  void compact;
}
